### ISSUE TITLE
Helper methods on `Transform`

### ISSUE THREAD

**Issue by lylythechosenone** [bevyengine/bevy#14426]
## What problem does this solve or what need does it fill?

Currently, working with a transform is cumbersome, requiring many lines to perform simple tasks. This can be seen easily in the [translation example](https://github.com/bevyengine/bevy/blob/main/examples/transforms/translation.rs):
```rust
let direction = transform.local_x();
transform.translation += direction * cube.speed * timer.delta_seconds();
```
Note that this has to be two lines instead of one, because of borrow checker issues.

## What solution would you like?

Simple methods like `translate`, `local_translate` would make such operations much simpler. The above could become
```rust
transform.local_translate(direction * cube.speed * timer.delta_seconds());
```
which is much simpler and easier to understand.

## What alternative(s) have you considered?

The original example does *work*, but as I said, it's quite cumbersome.

## Additional context

I know Unity has functions like these, although I don't know how many other engines do. Notably, Godot does not.

**Comment by alice-i-cecile** [comment]
Yep, if you have good ideas for helpers like these please PR them (ideally in a couple of small PRs) :D

**Comment by lylythechosenone** [comment]
Yeah, I can totally write them. Look out for a PR in the next day or so.

**Comment by Bayslak** [comment]
Hello, how are you doing? 😊

First time here and I wanted to work on this issue if it is still open!

**Comment by alice-i-cecile** [comment]
Yes, feel free @Bayslak :)
